gwcs/tests/conftest.py

```diff
@@ -0,0 +1,135 @@
+"""
+This file contains a set of pytest fixtures which are different gwcses for testing.
+"""
+import pytest
+
+import astropy.units as u
+from astropy import coordinates as coord
+from astropy.modeling import models
+from astropy.time import Time
+
+from .. import coordinate_frames as cf
+from .. import wcs
+
+# frames
+detector_2d = cf.Frame2D(name='detector', axes_order=(0, 1))
+icrs_sky_frame = cf.CelestialFrame(reference_frame=coord.ICRS(),
+                                   axes_order=(0, 1))
+
+freq_frame = cf.SpectralFrame(name='freq', unit=[
+    u.Hz,
+], axes_order=(2, ))
+wave_frame = cf.SpectralFrame(name='wave',
+                              unit=[
+                                  u.m,
+                              ],
+                              axes_order=(2, ),
+                              axes_names=('lambda', ))
+
+# transforms
+model_3d_shift = models.Shift(1) & models.Shift(2)
+model_1d_scale = models.Scale(2)
+
+
+@pytest.fixture
+def gwcs_2d_spatial_shift():
+    """
+    A simple one step spatial WCS, in ICRS with a 1 and 2 px shift.
+    """
+    pipe = [(detector_2d, model_3d_shift), (icrs_sky_frame, None)]
+
+    return wcs.WCS(pipe)
+
+
+@pytest.fixture
+def gwcs_1d_freq():
+    return wcs.WCS([(detector_2d, model_1d_scale), (freq_frame, None)])
```

Comment:
Well, it works on master.
I am working on other changes that will allow calling the WCS with non-coordinate inputs. To have a reasonable interface it would help if it is required that the number of inputs of the forward_transform equals the number of axes in the input_frame. Do you think this is too restrictive? The above use case works only because the model ignores any additional inputs. 
```
model_1d_scale(1,2,3,4)                                                                                                                           
Out[14]: 1.0
```
I think it may even be useful to catch this as an error in gwcs (or at least this is what I am doing in my workign copy).
```
tr_inputs = self.forward_transform.n_inputs
    if len(inputs) != tr_inputs:
            raise ValueError("The number of inputs {} does not match the "
            "number of inputs of the forward "
            "transforms {}".format(len(inputs), tr_inputs))
```
Is there a use case where this is undesirable?